### PR TITLE
Expenses List: Search Bar Addition

### PR DESCRIPTION
**Pull request by SudharakaP**
Add search bar in order to search for expenses.

Fixes https://github.com/opencollective/opencollective/issues/3214

### CODE REVIEW

File: pages/expenses.js [hunk 1]
```diff
@@ -84,11 +95,23 @@ class ExpensePage extends React.Component {
...
   buildFilterLinkParams(params) {
     return {
-      ...pick(this.props, ['collectiveSlug', 'limit', 'parentCollectiveSlug', 'tag', 'type']),
+      ...pick(this.props, ['collectiveSlug', 'limit', 'parentCollectiveSlug', 'tag', 'type', 'searchTerm']),
       ...params,
     };
   }
 
+  handleSearch(event) {
+    const searchInput = event.target.elements.q;
```
Comment (Betree) [inline pages/expenses.js:104]: Can't we directly get the value through `event.target.value`?

Comment (SudharakaP) [reply pages/expenses.js:104]: @Betree : No unfortunately; `event.target` is a `<form>` element and `event.target.value` will be undefined.

**Comment by SudharakaP** [comment]
> Nice! We're going to have a conflict after we merge #4440, but it shouldn't be too difficult to merge, both PRs are complementary.

@Betree : Rebase done and merge conflicts resolved. :+1:

File: pages/expenses.js (hunk 1 above)
Comment (Betree) [reply pages/expenses.js:141]: I see, thanks for the explanation. That's not a great API from `SearchForm` but it works

**Comment by SudharakaP** [comment]
> @SudharakaP It works really great, I wanted to add some feedback but with Github was down so I sent a commit directly. It's mainly about setting a default value for the input, making sure that we use the reference styles (by using a `StyledInput`), and i18n.
> 
> I think this one is ready to merge, I'll add some more comments on API now, we can improve some parts there.

@Betree : Wonderful and thanks a lot for the explanation. :+1:

**Pull request merged**